lib/operations/create_index.js

```diff
@@ -1,166 +1,130 @@
 'use strict';
 
-const Aspect = require('./operation').Aspect;
-const defineAspects = require('./operation').defineAspects;
-const OperationBase = require('./operation').OperationBase;
 const applyWriteConcern = require('../utils').applyWriteConcern;
+const Aspect = require('./operation').Aspect;
 const CommandOperation = require('./command');
 const CONSTANTS = require('../constants');
+const defineAspects = require('./operation').defineAspects;
 const handleCallback = require('../utils').handleCallback;
 const MongoError = require('../core').MongoError;
 const parseIndexOptions = require('../utils').parseIndexOptions;
 const ReadPreference = require('../core').ReadPreference;
 const toError = require('../utils').toError;
 
-class CreateIndexOperation extends OperationBase {
+const keysToOmit = new Set([
+  'name',
+  'key',
+  'writeConcern',
+  'w',
+  'wtimeout',
+  'j',
+  'fsync',
+  'readPreference',
+  'session'
+]);
+
+class CreateIndexOperation extends CommandOperation {
   constructor(db, name, fieldOrSpec, options) {
-    super(options);
+    // Build the index
+    const indexParameters = parseIndexOptions(fieldOrSpec);
```

Comment:
Can we put the construction of the indexes in a `buildCommand`?

Reply:
IMO it makes sense to keep the construction of the indexes here since we use `this.indexes` in `_buildCommand` and `execute`.

Reply:
I think Dan's right here. `this.indexes` is only used to actually build the command. So you'd want to just store things like `name` and `fieldOfSpec` on the operation, and move everything that needs `indexes` into `buildCommand`

Reply:
We also need `indexes` to check `indexes.collation` [here](https://github.com/mongodb/node-mongodb-native/pull/1996/files#diff-8385c257931c3c6f8cb44e7fa7b83f8aR72).

Reply:
I would argue the code you linked to should go into `execute` as well, especially since it does feature detection for the operation (this would eventually be done on a connection level), and optionally needs to return an async error to the caller.

Reply:
The linked code is in `execute`, which is why I think we need `this.indexes`.

Reply:
Ah yeah, that link jumped a little on the files page whoops 😄.  To solve the immediate problem we could just `indexes` into `_buildCommand` and that gets us out of building them in the constructor. But, perhaps more importantly, the check in `execute` is to see if `collation` has been specified. I think we're looking in the wrong place here, collation should be something specified by `options` or perhaps inherited from a parent object (database/collection).